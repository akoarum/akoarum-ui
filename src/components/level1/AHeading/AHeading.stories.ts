import { storiesOf } from '@storybook/vue'
import { withKnobs, number, text } from '@storybook/addon-knobs'
import { withInfo } from 'storybook-addon-vue-info'
import AHeading from './AHeading.vue'

storiesOf('level1/AHeading', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('VisualLevel1', () => ({
    components: { AHeading },
    props: {
      level: { default: number('要素レベル', 1) },
      texts: { default: text('テキスト', '吾輩は猫である') }
    },
    template: `
      <AHeading :level="level" :visual="1">{{ texts }}</AHeading>
    `
  }), { info: {} })
  .add('VisualLevel2', () => ({
    components: { AHeading },
    props: {
      level: { default: number('要素レベル', 1) },
      texts: { default: text('テキスト', '吾輩は猫である') }
    },
    template: `
      <AHeading :level="level" :visual="2">{{ texts }}</AHeading>
    `
  }), { info: {} })
  .add('VisualLevel3', () => ({
    components: { AHeading },
    props: {
      level: { default: number('要素レベル', 1) },
      texts: { default: text('テキスト', '吾輩は猫である') }
    },
    template: `
      <AHeading :level="level" :visual="3">{{ texts }}</AHeading>
    `
  }), { info: {} })
  .add('VisualLevel4', () => ({
    components: { AHeading },
    props: {
      level: { default: number('要素レベル', 1) },
      texts: { default: text('テキスト', '吾輩は猫である') }
    },
    template: `
      <AHeading :level="level" :visual="4">{{ texts }}</AHeading>
    `
  }), { info: {} })
  .add('VisualLevel5', () => ({
    components: { AHeading },
    props: {
      level: { default: number('要素レベル', 1) },
      texts: { default: text('テキスト', '吾輩は猫である') }
    },
    template: `
      <AHeading :level="level" :visual="5">{{ texts }}</AHeading>
    `
  }), { info: {} })
  .add('VisualLevel6', () => ({
    components: { AHeading },
    props: {
      level: { default: number('要素レベル', 1) },
      texts: { default: text('テキスト', '吾輩は猫である') }
    },
    template: `
      <AHeading :level="level" :visual="6">{{ texts }}</AHeading>
    `
  }), { info: {} })
