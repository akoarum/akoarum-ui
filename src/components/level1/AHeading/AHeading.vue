<script lang="ts">
import { CreateElement, VNode } from 'vue'
import { defineComponent, computed } from '@vue/composition-api'

interface Props {
  level: number
  visual: number
}

export default defineComponent({
  props: {
    level: { type: Number, default: 2 },
    visual: { type: Number, default: 2 }
  },
  setup(props: Props) {
    const styledClasses = computed<string[]>(() => {
      const classes: string[] = ['aHeading']
      classes.push(`-lv${props.visual}`)
      return classes
    })

    return { styledClasses }
  },
  render(createElement: CreateElement): VNode {
    return createElement(
      `h${this.level}`,
      {
        class: this.styledClasses
      },
      this.$slots.default
    )
  }
})
</script>

<style lang="scss" scoped>
.aHeading {
  &.-lv1 {
    font-size: 3.2rem;
    font-weight: bold;
    line-height: 1.2;
  }
  &.-lv2 {
    font-size: 2.9rem;
    font-weight: bold;
    line-height: 1.3;
  }
  &.-lv3 {
    font-size: 2.4rem;
    font-weight: bold;
    line-height: 1.4;
  }
  &.-lv4 {
    font-size: 2.1rem;
    font-weight: bold;
    line-height: 1.4;
  }
  &.-lv5 {
    font-size: 1.8rem;
    font-weight: bold;
    line-height: 1.6;
  }
  &.-lv6 {
    font-size: 1.6rem;
    font-weight: bold;
    line-height: 1.9;
  }
}
</style>
