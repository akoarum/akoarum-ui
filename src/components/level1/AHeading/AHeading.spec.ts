import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import AHeading from './AHeading.vue'

describe('AHeading', () => {
  const localVue = createLocalVue()
  let wrapper

  localVue.use(VueCompositionApi)

  beforeEach(() => {
    wrapper = shallowMount(AHeading, {
      propsData: {
        level: 3
      },
      slots: {
        default: '見出しテキスト'
      },
      localVue
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('levelで指定した要素レベルでレンダリングする', () => {
      expect(wrapper.is('h3')).toBe(true)
    })

    it('styledClasses: visualで指定したレベルの見た目になる', () => {
      wrapper.setProps({ visual: 4 })
      expect(wrapper.vm.styledClasses).toEqual(['aHeading', '-lv4'])
    })
  })
})
