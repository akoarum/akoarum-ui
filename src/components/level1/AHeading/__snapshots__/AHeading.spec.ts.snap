// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AHeading initialized snapshot 1`] = `
<h3
  class="aHeading -lv2"
>
  見出しテキスト
</h3>
`;
